### DISCUSSION TITLE
Is there a way to prune remote branches?

### DISCUSSION THREAD

**Discussion by gaogao-qwq** [jesseduffield/lazygit#3803] [Q&A] [answered]
My working gitlab repository will automatically remove the branches that are already merge into main branch. I couldn't found a keybind of prune remote branches, thus I have to run `git remote prune origin` manually.

<img width="1075" alt="image" src="https://github.com/user-attachments/assets/42a10900-26c2-478d-b478-a5d9fe74fb12">

**Comment by mark2185** [comment]
I usually have this config option:
```bash
$> git config remote.origin.prune true
```
And then you can go to the `Remotes` panel and just fetch `origin`.

Or create a custom command to run `git remote prune <currently selected remote>`.

**Comment by stefanhaller** [comment, marked as answer]
I recommend to do `git config --global fetch.prune true`. This way, remote branches are pruned automatically whenever you type `git fetch` or `git pull` on the command line; it also works in lazygit.

Luka's answer should also work, but needs to be configured for each remote separately.
